api/document/tests/json_importer/importer_test.py

```diff
@@ -13,3 +18,25 @@ def test_convert_paragraph_works():
     assert para.node_type == 'para'
     assert para.text == 'Hello there'
     assert para.json_content == para_primitive['content']
+
+
+@pytest.mark.django_db
+def test_import_json_doc_works():
+    policy = mommy.make(Policy)
+    para_primitive = f.para(content=[
+        f.text('Hello '),
+        f.external_link('http://example.org/', [f.text('there')]),
+    ])
+    para = import_json_doc(policy, para_primitive)
+
+    assert para.node_type == 'para'
+    assert para.text == 'Hello there'
+    annos = [anno for anno in para.model.annotations()]
+    assert len(annos) == 1
+    assert annos[0].href == 'http://example.org/'
+
+    para2 = import_json_doc(policy, para_primitive)
+    assert para.pk != para2.pk
+
+    with pytest.raises(DocNode.DoesNotExist):
+        assert para.model.refresh_from_db()
```

Comment:
This took me a few re-reads to understand. What do you think about
```python
assert not DocNode.objects.exists(pk=para.pk)
```

Reply:
Ah good idea! That didn't work (`exists` didn't like the keyword arg) but this did:

```python
assert not DocNode.objects.filter(pk=para.pk).exists()
```